integration_tests/src/main/python/json_test.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from asserts import assert_gpu_and_cpu_are_equal_collect
+from data_gen import *
+from pyspark.sql.types import *
+
+def mk_json_str_gen(pattern):
+    return StringGen(pattern).with_special_case('').with_special_pattern('.{0,10}')
+
+@pytest.mark.parametrize('json_str_pattern', ["\\{\"store\": \{\"fruit\": \[\\{" "\"weight\":\d,\"type\":\"[a-z]{1,9}\"\\}\], " \
+                   "\"bicycle\":\\{\"price\":\d\d\.\d\d,\"color\":\"[a-z]{0,4}\"\\}\\},\"email\":\"[" \
+                   "a-z]{1,5}\\@[a-z]{3,10}\.com\",\"owner\":\"[a-z]{3,8}\"\\}", "\\{\"a\": \"[a-z]{1,3}\"\\}"], ids=idfn)
```

Comment:
This would be a _lot_ clearer to read without all the escaped characters.  Recommend using raw strings here, something like this (untested):
```suggestion
@pytest.mark.parametrize('json_str_pattern', [r'\{"store": \{"fruit": \[\{"weight":\d,"type":"[a-z]{1,9}"\}\], ' \
                   r'"bicycle":\{"price":\d\d\.\d\d,"color":"[a-z]{0,4}"\}\},' \
                   r'"email":"[a-z]{1,5}\@[a-z]{3,10}\.com","owner":"[a-z]{3,8}"\}',
                   r'\{"a": "[a-z]{1,3}"\}'], ids=idfn)
```